packages/editor/src/components/post-preview-button/index.js

```diff
@@ -210,17 +211,23 @@ export default compose( [
 			getEditedPostPreviewLink,
 		} = select( 'core/editor' );
 		const {
+			getAutosave,
 			getPostType,
+			hasFetchedAutosave,
 		} = select( 'core' );
 
 		const previewLink = getEditedPostPreviewLink();
 		const postType = getPostType( getEditedPostAttribute( 'type' ) );
+		const post = getCurrentPost();
```

Comment:
Another reason simple arguments to selectors would be preferable is in considering that a selector like this would could potentially be quite non-performant in aggregating the properties of the object of a "post", whereas getting only its `id` would most certainly be trivial in comparison.

`getCurrentPost` is actually trivial, but something like `getBlock` is a good example of the opposite.